trade-finance-manager-api/src/v1/rest-mappings/mappings/facilities/mapFacilityType.ts

```diff
@@ -0,0 +1,54 @@
+import { Facility, FacilityType, FACILITY_TYPE, MAPPED_FACILITY_TYPE } from '@ukef/dtfs2-common';
+
+const { BOND, LOAN, CASH, CONTINGENT } = FACILITY_TYPE;
+
+/**
+ * Maps type to a different facility type used in TFM-UI and TFM-API.
+ * @param type - the facility type in the database
+ * @param facilitySnapshot - the facility snapshot from the database
+ * @returns the mapped facility type
+ *
+ * @example
+ * // returns 'Bid Bond'
+ * mapFacilityType('Bond', {...facilitySnapshot, code: 'BSS', bondType: 'Bid Bond'})
+ *
+ * @example
+ * // returns 'Loan'
+ * mapFacilityType('Loan', facilitySnapshot)
+ */
+export const mapFacilityType = (type: FacilityType, facilitySnapshot: Facility): string | null => {
+  if (type === BOND) {
+    return facilitySnapshot.bondType ?? null;
+  }
+
+  if (type === LOAN) {
+    return MAPPED_FACILITY_TYPE.LOAN;
+  }
+
+  return null;
+};
+
+/**
+ * Maps a GEF facility type to a different facility type used in TFM-UI and TFM-API.
+ * @param type - the facility type in the database
+ * @returns the mapped facility type
+ *
+ * @example
+ * // returns 'Cash Facility'
+ * mapGefFacilityType('Cash')
+ *
+ * @example
+ * // returns 'Contingent Facility'
+ * mapGefFacilityType('Contingent')
+ */
+export const mapGefFacilityType = (type: FacilityType): string | null => {
```

Comment:
Brand new file showing in the diff as I've changed it from JS to TS. I've split out the Gef facility type mappings as GEF facilities will only ever have Cash or Contingent types